src/server/transport.rs

```diff
@@ -109,18 +112,30 @@ impl RaftStoreRouter for ServerRaftStoreRouter {
 #[derive(Clone)]
 pub struct ServerTransport {
     ch: SendCh<Msg>,
+    raft_client: Arc<RwLock<RaftClient>>,
 }
 
 impl ServerTransport {
-    pub fn new(ch: SendCh<Msg>) -> ServerTransport {
-        ServerTransport { ch: ch }
+    pub fn new(ch: SendCh<Msg>, raft_client: Arc<RwLock<RaftClient>>) -> ServerTransport {
+        ServerTransport {
+            ch: ch,
+            raft_client: raft_client,
+        }
     }
 }
 
 impl Transport for ServerTransport {
     fn send(&self, msg: RaftMessage) -> RaftStoreResult<()> {
         let to_store_id = msg.get_to_peer().get_store_id();
-
+        if !msg.get_message().has_snapshot() {
+            let addr = self.raft_client.rl().addrs.get(&to_store_id).map(|x| x.to_owned());
+            if let Some(addr) = addr {
+                if let Err(e) = self.raft_client.wl().send(to_store_id, addr.to_owned(), msg) {
```

Comment:
should consider the case of address outdated if the remote store changes its address, although this may be a rare case. 

Or if we can't connect remote store many times, do we need to get the address from PD again?